src/parameterizations/lateral/MOM_hor_visc.F90

```diff
@@ -1426,52 +1441,40 @@ subroutine horizontal_viscosity(u, v, h, diffu, diffv, MEKE, VarMix, G, GV, US,
     endif
 
     if (CS%use_GME) then
-      !### This call to get the 3-d GME diffusivity arrays and the subsequent blocking halo update
-      !    should occur outside of the k-loop, and perhaps the halo update should occur outside of
-      !    this routine altogether!
-      call thickness_diffuse_get_KH(TD, KH_u_GME, KH_v_GME, G, GV)
-      call pass_vector(KH_u_GME, KH_v_GME, G%Domain)
-
-      do j=Jsq,Jeq+1 ; do i=Isq,Ieq+1
+      do j=Jsq-1,Jeq+2 ; do i=Isq-1,Ieq+2
         GME_coeff = GME_effic_h(i,j) * 0.25 * &
             ((KH_u_GME(I,j,k)+KH_u_GME(I-1,j,k)) + (KH_v_GME(i,J,k)+KH_v_GME(i,J-1,k)))
         GME_coeff = MIN(GME_coeff, CS%GME_limiter)
 
         if ((CS%id_GME_coeff_h>0) .or. find_FrictWork) GME_coeff_h(i,j,k) = GME_coeff
         str_xx_GME(i,j) = GME_coeff * sh_xx_bt(i,j)
-
       enddo ; enddo
 
-      do J=js-1,Jeq ; do I=is-1,Ieq
+      do J=Jsq-2,Jeq+1 ; do I=Isq-2,Ieq+1
```

Comment:
These q-point loop ranges should have symmetric halos with respect to the q-point computational domain or non-symmetric halos with respect to the h-points.  I.e., one of `do J=Jsq-1,Jeq+1 ; do I=Isq-1,Ieq+1` or `do J=js-2,je+1 ; do I=is-2,ie+1` should be right, depending on how str_xy_GME is used later.